tests/pass/shims/time-with-isolation.rs

```diff
@@ -25,11 +25,20 @@ fn test_time_passes() {
     // The virtual clock is deterministic and I got 29us on a 64-bit Linux machine. However, this
     // changes according to the platform so we use an interval to be safe. This should be updated
     // if `NANOSECONDS_PER_BASIC_BLOCK` changes.
-    assert!(diff.as_micros() > 10);
-    assert!(diff.as_micros() < 40);
+    assert!(diff.as_millis() > 5);
+    assert!(diff.as_millis() < 20);
+}
+
+fn test_block_for_one_second() {
+    let end = Instant::now() + Duration::from_secs(1);
+    // This takes a long time, as we only increment when statements are executed.
+    // When we sleep on all threads, we fast forward to the sleep duration, which we can't
+    // do with busy waiting.
+    while Instant::now() < end {}
```

Comment:
Oh this is an interesting test. It's not quite what I had in mind when talking about syncing up the two clocks.

What happens for a program like this?
```rust
let begin = Instant::now();
for _ in 0..100_000 {}
let time = begin.elapsed();
println!("The loop took {time:?}");
```
What does this print in regular execution (with debug mode and `-Zmir-opt-level=0`), vs Miri with isolation?

Reply:
18s under isolation, between 1 and 2 seconds on regular execution, opt level has no measurable effect.

Reply:
> 36ms under isolation (on master)

Okay yeah the current number is way too small then.^^

Looks like this PR is somewhat overshooting though. Should we go for 1000ns maybe?

Reply:
Whatever we choose, it will be odd as it depends on basic block counts which have no mapping to actual cost. Overshooting will make tests like the one I wrote finish faster, while producing somewhat odd values for tests like the one you wrote, without actually affecting the host wall time of miri interpreting your test.

Reply:
How long does your test take with 1000ns?

Reply:
5s

Reply:
Factor 5 on both ends, nice.

:shrug: no strong opinion, if you want to stick with 5 microseconds then I won't fight that.
The PR has some other unresolved threads though. :)